### PR TITLE
Add table aws_iam_user_ssh_key. closes #1339

### LINKED ISSUE #1339: Add table aws_iam_user_ssh_key
**References**
https://docs.aws.amazon.com/IAM/latest/APIReference/API_ListServiceSpecificCredentials.html

**Use Case**
as of now is not possible to list if IAM users have "SSH keys for AWS CodeCommit" enabled
Should work similar to aws_iam_access_key

### PR DESCRIPTION
**Pull request by luisffc**
Addresses: #1339

TODO:

- [x] Write integration tests

# Integration test logs
<details>
  <summary>Logs</summary>

```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_iam_ssh_public_key []

PRETEST: tests/aws_iam_ssh_public_key

TEST: tests/aws_iam_ssh_public_key
Running terraform
data.aws_region.alternate: Reading...
data.aws_caller_identity.current: Reading...
data.aws_partition.current: Reading...
data.aws_region.primary: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-2]
data.aws_region.primary: Read complete after 0s [id=us-east-1]
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_caller_identity.current: Read complete after 1s [id=123456789012]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_iam_user.named_test_resource will be created
  + resource "aws_iam_user" "named_test_resource" {
      + arn           = (known after apply)
      + force_destroy = false
      + id            = (known after apply)
      + name          = "turbottest69646"
      + path          = "/"
      + tags          = {
          + "name" = "turbottest69646"
        }
      + tags_all      = {
          + "name" = "turbottest69646"
        }
      + unique_id     = (known after apply)
    }

  # aws_iam_user_ssh_key.named_test_resource will be created
  + resource "aws_iam_user_ssh_key" "named_test_resource" {
      + encoding          = "SSH"
      + fingerprint       = (known after apply)
      + id                = (known after apply)
      + public_key        = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQD3F6tyPEFEzV0LX3X8BsXdMsQz1x2cEikKDEY0aIj41qgxMCP/iteneqXSIFZBp5vizPvaoIR3Um9xK7PGoW8giupGn+EPuxIA4cDM4vzOqOkiMPhz5XK0whEjkVzTo4+S0puvDZuwIsdiW9mxhJc7tgBNL0cYlWSYVkz4G/fslNfRPW5mYAM49f4fhtxPb5ok4Q2Lg9dPKVHO/Bgeu5woMc7RY0p1ej6D4CKFE6lymSDJpW0YHX/wqE9+cfEauh7xZcG0q9t2ta6F6fmX0agvpFyZo8aFbXeUBr7osSCJNgvavWbM/06niWrOvYX2xwWdhXmXSrbX8ZbabVohBK41 mytest@mydomain.com"
      + ssh_public_key_id = (known after apply)
      + status            = (known after apply)
      + username          = "turbottest69646"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id        = "123456789012"
  + aws_partition     = "aws"
  + resource_name     = "turbottest69646"
  + ssh_public_key_id = (known after apply)
aws_iam_user.named_test_resource: Creating...
aws_iam_user.named_test_resource: Creation complete after 2s [id=turbottest69646]
aws_iam_user_ssh_key.named_test_resource: Creating...
aws_iam_user_ssh_key.named_test_resource: Creation complete after 1s [id=APKAZGW7IOFIOOVJBBUT]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 43, in data "null_data_source" "resource":
  43: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

account_id = "123456789012"
aws_partition = "aws"
resource_name = "turbottest69646"
ssh_public_key_id = "APKAZGW7IOFIOOVJBBUT"

Running SQL query: test-get-query.sql
[
  {
    "ssh_public_key_id": "APKAZGW7IOFIOOVJBBUT",
    "user_name": "turbottest69646"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "ssh_public_key_id": "APKAZGW7IOFIOOVJBBUT",
    "user_name": "turbottest69646"
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "region": "global",
    "title": "APKAZGW7IOFIOOVJBBUT"
  }
]
✔ PASSED

POSTTEST: tests/aws_iam_ssh_public_key

TEARDOWN: tests/aws_iam_ssh_public_key

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_iam_ssh_public_key
+----------------------+------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------->
| ssh_public_key_id    | ssh_public_key_body_pem                                          | ssh_public_key_body_rsa                                                                                      >
+----------------------+------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------->
| APKAZGW7IOFIE6RBHFUL | -----BEGIN PUBLIC KEY-----                                       | ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQD3F6tyPEFEzV0LX3X8BsXdMsQz1x2cEikKDEY0aIj41qgxMCP/iteneqXSIFZBp5vizPvao>
|                      | MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA9xercjxBRM1dC191/AbF |                                                                                                              >
|                      | 3TLEM9cdnBIpCgxGNGiI+NaoMTAj/4rXp3ql0iBWQaeb4sz72qCEd1JvcSuzxqFv |                                                                                                              >
|                      | IIrqRp/hD7sSAOHAzOL8zqjpIjD4c+VytMIRI5Fc06OPktKbrw2bsCLHYlvZsYSX |                                                                                                              >
|                      | O7YATS9HGJVkmFZM+Bv37JTX0T1uZmADOPX+H4bcT2+aJOENi4PXTylRzvwYHruc |                                                                                                              >
|                      | KDHO0WNKdXo+g+AihROpcpkgyaVtGB1/8KhPfnHxGroe8WXBtKvbdrWuhen5l9Go |                                                                                                              >
|                      | L6RcmaPGhW13lAa+6LEgiTYL2r1mzP9Op4lqzr2F9scFnYV5l0q21/GW2m1aIQSu |                                                                                                              >
|                      | NQIDAQAB                                                         |                                                                                                              >
|                      | -----END PUBLIC KEY-----                                         |                                                                                                              >
|                      |                                                                  |                                                                                                              >
+----------------------+------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------
```
</details>

### CODE REVIEW

**Comment by rajlearner17** [comment]
@ParthaI Pls re-work on this to map the new changes in 0.93.0

**Comment by cbruno10** [comment]
@rajlearner17 Please tag me as reviewer once this is ready, the SSH key tables are tricky due to different formats, so I'd like to review.

File: aws/table_aws_iam_user_ssh_key.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+		},
+		Columns: awsGlobalRegionColumns([]*plugin.Column{
+			{
+				Name:        "ssh_public_key_id",
+				Description: "The unique identifier for the SSH public key.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("SSHPublicKeyId"),
+			},
+			{
+				Name:        "ssh_public_key_body_pem",
```
Comment (cbruno10) [inline aws/table_aws_iam_user_ssh_key.go:41]: This column is one I'm unsure about, should we show it as this format?

Comment (ParthaI) [reply aws/table_aws_iam_user_ssh_key.go:41]: @cbruno10 The API [GetSSHPublicKey](https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetSSHPublicKey.html) provides the public key body with both format(`SSH` and `PEM`). From this API there is property called `SSHPublicKeyBody` which provides the key body by providing the `Encoding` value(`SSH` or `PEM`) in the API parameter. So we have added both of the column (`ssh_public_key_body_pem` and `ssh_public_key_body_rsa`) in this table.

Comment (ParthaI) [reply aws/table_aws_iam_user_ssh_key.go:41]: @cbruno10 We may have similar scenarios where we can upload the same SSH public key in different formats to enable my access i.e. I can query the column and get the Public key > provide the same in other application.
image

**Comment by misraved** [comment]
Closing this PR since it is not planned. We will reopen this PR incase we have a strong use case for adding this table 👍.

**Pull request closed without merging**